### DISCUSSION TITLE
Any idea how I can use this with next-urql?

### DISCUSSION THREAD

**Discussion by Axedyson** [i18next/next-i18next#1842] [Q&A] [answered]
[next-urql](https://github.com/FormidableLabs/urql/tree/main/packages/next-urql) requires one to use getInitialProps so it's quite hard to use it with this library

**Comment by Axedyson** [comment, marked as answer]
Okay the solution is simply to export a non-getInitialProps data fetching next.js function and use that instead of getInitialProps whenever I have to do data fetching during SSR. There is now more code to maintain but I guess this is the only solution

**Reply by jmcoder1** [reply]
Can you show what this looks like?

**Reply by jmcoder1** [reply]
I figured out you can do this with Urql quite easily. 

Here's a guide: https://formidable.com/open-source/urql/docs/advanced/server-side-rendering/#ssr-with-getstaticprops-or-getserversideprops
